party_menu/party_menu.lua

```diff
@@ -0,0 +1,521 @@
+-- TODO: Replace command.input('/echo') with chat.add_text() when possible
+-- TODO: Consider adding a title row that describes each column to reduce clutter.
+-- TODO: Consider adding distance to player as a column
+-- TODO: Consider adding debuff image functionality.
+-- TODO: Consider adding settings to show different data on player vs partymembers
+-- TODO: Optimize/revisit incoming packet code, in particular look into getting party zone ids to update correctly when party members zone, for now just display 'Not in zone'
+
+-- Known Issues:
+-- TODO: Party member will briefly show as 'DEAD' when zoning into the same zone as the player.
+-- TODO: HP updates can be a bit slow, particularly after a cure.
+
+local player = require('player')
+local ui = require('ui')
+local string = require('string')
+local res = require('resources')
+local status_effects = require('status_effects')
+local packets = require('packets')
+local chat = require('chat')
+local command = require('command')
+local settings = require('settings')
+local target = require('target')
+local party = require('party')
+local math = require('math')
+local os = require('os')
+
+local debuff_ids = require('debuff_ids')
+
+local hp_modes = {'missinghp', 'hp', 'hp/hp_max', 'hide'}
+local mp_modes = {'mp', 'mp/mp_max', 'hide'}
+
+local zoning = false
+local zone_out_time = os.clock()
+local zone_in_delay = 10
+
+local defaults = {
+    names_visible = true,
+    hp_mode_index = 1,
+    mp_mode_index = 3,
+    tp_visible = false,
+    debuffs_visible = true,
+    dynamic_coloring = true,
+    text_size = 20,
+    text_typeface = 'Consolas',
+    global_window_state = {
+        style = 'chromeless',
+        color = ui.color.black,
+        closable = true,
+        x = 20,
+        y = 100,
+        width = 500,
+    }
+}
+local options = settings.load(defaults)
+
+local function get_party_size()
+    for i = 1, 6 do
+        if not party[i] then
+            return (i - 1)
+        end
+    end
+    return 6
+end
+
+local global_window_state = {}
+local function quick_shallow_copy()
```

Comment:
Yes, in particular I realized I needed to avoid overwriting the y position of the global window except when issuing the command to change the window position. Otherwise I found the overall window would position would not stay stationary when reloading the addon